File: src/main/java/edu/wpi/teamname/state/HomeState.java
package edu.wpi.teamname.state;

import javafx.beans.property.SimpleBooleanProperty;
import javafx.beans.property.SimpleStringProperty;
import lombok.Data;

@Data
public class HomeState {
  /* This stores the state of the home screen */

  private SimpleBooleanProperty clickedProperty = new SimpleBooleanProperty(false);
  private SimpleStringProperty currentNodeProperty = new SimpleStringProperty(null);

  @Override
  public String toString() {
    return super.toString();
  }
}

File: src/main/kotlin/edu/wpi/teamname/state/HomeState.kt
package edu.wpi.teamname.state

import javafx.beans.property.SimpleBooleanProperty
import javafx.beans.property.SimpleStringProperty

/* This stores the state of the home screen */
class HomeState {
    val visible = SimpleBooleanProperty(false)
    val text = SimpleStringProperty(null)
    var numClicks = -1
        private set

    fun toggleClicked() {
        visible.value = !visible.value
        if (visible.value) numClicks++
    }

    fun setNode(newVal: String?) {
        text.value = newVal
    }
}
